api/v1beta1/cluster_types.go

```diff
@@ -772,18 +778,21 @@ type MachinePoolTopology struct {
 	// class is the name of the MachinePoolClass used to create the pool of worker nodes.
 	// This should match one of the deployment classes defined in the ClusterClass object
 	// mentioned in the `Cluster.Spec.Class` field.
+	// +kubebuilder:validation:MaxLength=256
 	Class string `json:"class"`
 
 	// name is the unique identifier for this MachinePoolTopology.
 	// The value is used with other unique identifiers to create a MachinePool's Name
 	// (e.g. cluster's name, etc). In case the name is greater than the allowed maximum length,
 	// the values are hashed together.
+	// +kubebuilder:validation:MaxLength=256
```

Comment:
253 max as it is a subset of a CR name

Reply:
I'll resolve and apply the outcome of the MD discussion to both